File: LAFMAP9.1/app/src/main/java/com/example/lafmap/DatabaseHelper.java
package com.example.lafmap;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import java.util.ArrayList;
import java.util.List;

public class DatabaseHelper extends SQLiteOpenHelper {

    private static final String DATABASE_NAME = "LafMap.db";
    private static final int DATABASE_VERSION = 1;

    public static final String TABLE_NAME = "adverts";
    public static final String COL_ID = "id";
    public static final String COL_TYPE = "type";  // lost or found
    public static final String COL_NAME = "name";
    public static final String COL_PHONE = "phone";
    public static final String COL_DESCRIPTION = "description";
    public static final String COL_DATE = "date";
    public static final String COL_LOCATION = "location";

    public DatabaseHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        String createTable = "CREATE TABLE " + TABLE_NAME + " (" +
                COL_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                COL_TYPE + " TEXT," +
                COL_NAME + " TEXT," +
                COL_PHONE + " TEXT," +
                COL_DESCRIPTION + " TEXT," +
                COL_DATE + " TEXT," +
                COL_LOCATION + " TEXT)";
        db.execSQL(createTable);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
        onCreate(db);
    }

    public boolean insertAdvert(String type, String name, String phone, String description, String date, String location) {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put(COL_TYPE, type);
        values.put(COL_NAME, name);
        values.put(COL_PHONE, phone);
        values.put(COL_DESCRIPTION, description);
        values.put(COL_DATE, date);
        values.put(COL_LOCATION, location);

        long result = db.insert(TABLE_NAME, null, values);
        return result != -1;
    }

    public List<Advert> getAllAdverts() {
        List<Advert> adverts = new ArrayList<>();
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor = db.rawQuery("SELECT * FROM " + TABLE_NAME + " ORDER BY " + COL_ID + " DESC", null);

        if (cursor.moveToFirst()) {
            do {
                int id = cursor.getInt(cursor.getColumnIndexOrThrow(COL_ID));
                String type = cursor.getString(cursor.getColumnIndexOrThrow(COL_TYPE));
                String name = cursor.getString(cursor.getColumnIndexOrThrow(COL_NAME));
                String phone = cursor.getString(cursor.getColumnIndexOrThrow(COL_PHONE));
                String description = cursor.getString(cursor.getColumnIndexOrThrow(COL_DESCRIPTION));
                String date = cursor.getString(cursor.getColumnIndexOrThrow(COL_DATE));
                String location = cursor.getString(cursor.getColumnIndexOrThrow(COL_LOCATION));

                adverts.add(new Advert(id, type, name, phone, description, date, location));
            } while (cursor.moveToNext());
        }
        cursor.close();
        return adverts;
    }
}

File: learningapptwo/app/src/main/java/com/example/learningapptwo/DatabaseHelper.kt
package com.example.learningapptwo

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

// Database helper to manage user data storage
class DatabaseHelper(context: Context) :
    SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    companion object {
        private const val DATABASE_NAME = "userDatabase"
        private const val DATABASE_VERSION = 2 // Incremented version to add the interests table
        private const val TABLE_NAME = "users"
        private const val COLUMN_USERNAME = "username"
        private const val COLUMN_PASSWORD = "password"
        private const val COLUMN_INTEREST = "interest" // Column to store selected interests
    }

    override fun onCreate(db: SQLiteDatabase?) {
        val CREATE_USER_TABLE = "CREATE TABLE $TABLE_NAME (" +
                "$COLUMN_USERNAME TEXT," +
                "$COLUMN_PASSWORD TEXT," +
                "$COLUMN_INTEREST TEXT)"
        db?.execSQL(CREATE_USER_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        if (oldVersion < newVersion) {
            db?.execSQL("DROP TABLE IF EXISTS $TABLE_NAME")
            onCreate(db)
        }
    }

    // Function to add a new user to the database
    fun addUser(username: String, password: String) {
        val db = writableDatabase
        val query = "INSERT INTO $TABLE_NAME ($COLUMN_USERNAME, $COLUMN_PASSWORD) VALUES (?, ?)"
        val statement = db.compileStatement(query)
        statement.bindString(1, username)
        statement.bindString(2, password)
        statement.executeInsert()
        db.close()
    }

    // Function to add selected interests for a user
    fun addInterests(username: String, interests: String) {
        val db = writableDatabase
        val query = "UPDATE $TABLE_NAME SET $COLUMN_INTEREST = ? WHERE $COLUMN_USERNAME = ?"
        val statement = db.compileStatement(query)
        statement.bindString(1, interests)
        statement.bindString(2, username)
        statement.executeUpdateDelete()
        db.close()
    }
}
